lib/omniauth/strategies/openid_connect.rb

```diff
@@ -312,13 +317,36 @@ def id_token_callback_phase
         call_app!
       end
 
+      def type_response?(type)
```

Comment:
What is the purpose of this method? I'm curious why did you choose this name "type_response?"

Reply:
Previously I had `id_token_response?` and I changed it to more generic 
```type_response?(type)```. Basically, it checks if `type` is specified in `options.response_type`.

We can check the presence of `code` only if `code` has been requested.
We can check the presence of `id_token` only if `id_token` has been requested.
We can verify `id_token` only if `id_token` has been requested.
We can request for access tokens only if `code` has been requested.

That's why we need to check it.

The name of this method can be improved.

Reply:
Gotcha, what would you say if we rename the method to something like `configured_response_type?(type)` or `known_response_type?(type)` ? I think it'd make it more clear what's the purpose of this method.